File: src/inference.py
def foo(x: int, y: int) -> int:
    return x + y

class C:
    def __init__(self, x: int):
        self.x = x
        
    def get_x(self) -> int:
        return self.x

foo(2, 3)

c = C(123)
print(c.get_x())

File: src/inference.rs
fn foo(x : i32, y : i32) -> i32 {
    x + y
}

struct C {
    x : i32
}

impl C {
    fn new(x : i32) -> Self {
        Self{x}
    }

    fn get_x(&self) -> i32 {
        self.x
    }
}

fn main() {
    println!("{:?}", foo(2, 3));

    let c = C::new(123);
    println!("{:?}", c.get_x());
}
